File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/synccenterdialog/SyncCenterDialogMvpView.java
package com.mifos.mifosxdroid.dialogfragments.synccenterdialog;

import com.mifos.mifosxdroid.base.MvpView;

/**
 * Created by mayankjindal on 10/07/17.
 */

public interface SyncCenterDialogMvpView extends MvpView {

    void showUI();

    void showSyncingCenter(String centerName);

    void showSyncedFailedCenters(int failedCount);

    void setMaxSingleSyncCenterProgressBar(int total);

    void setGroupSyncProgressBarMax(int count);

    void updateGroupSyncProgressBar(int i);

    void setClientSyncProgressBarMax(int count);

    void updateClientSyncProgressBar(int i);

    void updateSingleSyncCenterProgressBar(int i);

    void updateTotalSyncCenterProgressBarAndCount(int i);

    int getMaxSingleSyncCenterProgressBar();

    void showNetworkIsNotAvailable();

    void showCentersSyncSuccessfully();

    Boolean isOnline();

    void dismissDialog();

    void showDialog();

    void hideDialog();

    void showError(int s);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/synccenterdialog/SyncCenterDialogMvpView.kt
package com.mifos.mifosxdroid.dialogfragments.synccenterdialog

import com.mifos.mifosxdroid.base.MvpView

/**
 * Created by mayankjindal on 10/07/17.
 */
interface SyncCenterDialogMvpView : MvpView {
    fun showUI()
    fun showSyncingCenter(centerName: String)
    fun showSyncedFailedCenters(failedCount: Int)
    fun setGroupSyncProgressBarMax(count: Int)
    fun updateGroupSyncProgressBar(index: Int)
    fun setClientSyncProgressBarMax(count: Int)
    fun updateClientSyncProgressBar(index: Int)
    fun updateSingleSyncCenterProgressBar(count: Int)
    fun updateTotalSyncCenterProgressBarAndCount(index: Int)
    var maxSingleSyncCenterProgressBar: Int
    fun showNetworkIsNotAvailable()
    fun showCentersSyncSuccessfully()
    val isOnline: Boolean?
    fun dismissDialog()
    fun showDialog()
    fun hideDialog()
    fun showError(s: Int)
}